store/postgres/src/block_store.rs

```diff
@@ -456,9 +456,15 @@ impl BlockStore {
                     continue;
                 };
             }
+            match store.chain_head_block(&&store.chain).unwrap_or(None) {
+                Some(head) => {
+                    let lower_bound = head - ENV_VARS.reorg_threshold;
```

Comment:
Check for overflow. Also I'd give some slack here so we don't have to worry about off by one, race conditions or whatever, maybe use `2 * ENV_VARS.reorg_threshold`.